config/types_test.go

```diff
@@ -91,7 +91,12 @@ func TestTOMLParsingStringDurations(t *testing.T) {
 		"42m",
 		"100ms",
 		"100us",
-		"100ns"
+		"100ns",
+		"1d",
+		"7.5d",
+		"7d8h15m",
+		"3d7d",
+		"15m8h3.5d"
```

Comment:
I don't think these formats should be supported?

Reply:
Why? Golang's `ParseDuration` supports them, so why shouldn't we?

Reply:
It makes it more complex and was only intended to be a temporary workaround.. Or do you think differently?

Reply:
Yes I do think differently as I do not think this is in any way more complex than the other features. If the code change is too complex in your view, we should completely drop the feature as it won't become any simpler...

Reply:
Okay, Neverland then. 

But do add depreciation warnings.

Reply:
Or aren't we going to deprecate it?

Reply:
I do not see what we do deprecate.... `0d` is explicitly supported in the code and the remaining never worked. So either we add this or leave the stuff with an (overdue) error...

Reply:
The whole idea was initially to support it, but immediately deprecate it so it can be removed in next major version.

Its fine for me to not deprecate it and keep it working, but then the docs need to be updated.